### PR TITLE
Jsonize Pet Food

### PR DESCRIPTION
**Pull request by Saicchi**
#### Summary
Infrastructure "Jsonize Pet Food"

#### Purpose of change
Player is currently locked at 4 categories of pet food: `DOGFOOD`, `CATFOOD`, `CATTLEFODDER` and `BIRDFOOD`.

Allow different types of pet food to tame any monster if they have the matching food type.

#### Describe the solution

Add a new field for comestibles:
`"petfood": [ "FUNGALFRUIT", "MIGOFOOD" ]`

Add a new field for monsters:
```c++
"petfood": {
    "food": [ "CATFOOD", "YULECATFOOD" ], // food categories this monster accepts
    "feed": "The gigantic %s decides not to maul you today.", // (optional) message when feeding the monster the food
    "pet": "The %s is enjoying hunting the red laser dot." // (optional) message when playing with pet
}
```

If both of these categories match the player can feed the item to the monster.

#### Describe alternatives you've considered
Add a new bool to allow / prevent playing with the pet.

Make `feed` and `pet` a list of strings which gets randomly chosen when they're needed.

#### Testing
- Optional parameters
  - [x] feed
  - [x] pet
- [x] Player does not eat the item they're trying to feed.
- [x] Feeding animals
- [x] Feeding the "dog"
- [x] Loading Aftershock and DinoMod
- [x] Loading Dark Skies Above

I'm not testing `Desert Region` as the mod is marked as obsolete.

#### Additional context
Can't wait until we see zombie dog pet food.

Since I removed the `BIRDFOOD`, `CATFOOD` and `CATTLEFODDER` monster flags, monsters now need the `CANPLAY` flag if you wan't to play with them. Note that I did not remove `DOGFOOD` since it is still used with `iuse::dog_whistle`.

**Click here to view table with information about monsters**
<details>
  <summary>Pets</summary>

| Flag         | Mod           | Monster                       | Can Play | Feed                                 | Pet                                      |
| :---:        | :---          | :---                          | :---: | :---                                 | :---                                     |
| DOGFOOD      | DDA           | mon_dog                       | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_bull                  | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_bull_pup              | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_pitbullmix            | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_pitbullmix_pup        | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_beagle                | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_beagle_pup            | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_bcollie               | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_bcollie_pup           | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_boxer_pup             | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_chihuahua             | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_chihuahua_pup         | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_dachshund             | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_dachshund_pup         | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_gshepherd             | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_gshepherd_pup         | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_gpyrenees             | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_gpyrenees_pup         | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_rottweiler            | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_rottweiler_pup        | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_auscattle             | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_auscattle_pup         | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | DDA           | mon_dog_mutant_mongrel        | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| DOGFOOD      | Aftershock    | afs_mon_cyber_mastiff_pup     | Yes | The %s seems to like you!  It lets you pat its head and seems friendly. | The %s happily wags its tail while you pat their head. |
| CATFOOD      | DDA           | mon_cat_kitten                | Yes | The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines. | The %s sofly purrs as you pet it.  It demands more. |
| CATFOOD      | DDA           | mon_cat_kitten                | Yes | The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines. | The %s sofly purrs as you pet it.  It demands more. |
| CATFOOD      | DDA           | mon_cat                       | Yes | The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines. | The %s sofly purrs as you pet it.  It demands more. |
| CATFOOD      | DDA           | mon_cat_mutant_prism          | Yes | The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines. | The %s sofly purrs as you pet it.  It demands more. |
| CATFOOD      | DDA           | mon_cat_mutant_kitten_prism   | Yes | The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines. | The %s sofly purrs as you pet it.  It demands more. |
| BIRDFOOD     | DDA           | mon_chicken                   | Yes | The %s seems to like you!  It runs around your legs and seems friendly. | The %s runs around your leg.             |
| BIRDFOOD     | DDA           | mon_chicken_chick             | Yes | The %s seems to like you!  It runs around your legs and seems friendly. | The %s runs around your leg.             |
| BIRDFOOD     | DDA           | mon_crow_chick                | Yes | The %s seems to like you!  It runs around your legs and seems friendly. | The %s runs around your leg.             |
| BIRDFOOD     | DDA           | mon_duck_chick                | Yes | The %s seems to like you!  It runs around your legs and seems friendly. | The %s runs around your leg.             |
| BIRDFOOD     | DDA           | mon_pigeon                    | Yes | The %s seems to like you!  It runs around your legs and seems friendly. | The %s runs around your leg.             |
| BIRDFOOD     | DSA           | dks_mon_wisp                  | Yes | The %s seems to like you!  It runs around your legs and seems friendly. | The %s runs around your leg.             |
| CATTLEFODDER | DDA           | mon_cow_calf                  | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | DDA           | mon_cow                       | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | DDA           | mon_horse                     | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | DDA           | mon_pig_piglet                | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | DDA           | mon_rabbit                    | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | DDA           | mon_sheep                     | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | DDA           | afs_mon_rabbit                | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | DDA           | dks_mon_mossliz               | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | Desert Region | mon_bushelephant              | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | Desert Region | mon_bushelephant_fungal       | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| CATTLEFODDER | Magiclysm     | mon_nightmare_sheep           | No  | The %s seems to like you!  It lets you pat its head and seems friendly. |                                          |
| DINOFOOD_A   | DinoMod       | mon_eoraptor_hatchling        | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_compsognathus             | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_dimorphodon               | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_quetzalcoatlus            | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_compsognathus_hatchling   | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_mosasaurus_hatchling      | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_dilophosaurus_juvenile    | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_ceratosaurus_juvenile     | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_spinosaurus_juvenile      | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_allosaurus_juvenile       | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_acrocanthosaurus_juvenile | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_siats_juvenile            | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_albertosaurus_juvenile    | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_qianzhousaurus_juvenile   | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_tyrannosaurus_juvenile    | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_deinonychus_juvenile      | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_utahraptor_juvenile       | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_pteranodon_juvenile       | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_quetzalcoatlus_juvenile   | Yes |                                      |                                          |
| DINOFOOD_B   | DinoMod       | mon_mosasaurus_juvenile       | Yes |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_gallimimus                | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_nothronychus              | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_scutellosaurus            | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_stegosaurus               | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_dyoplosaurus              | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_ankylosaurus              | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_nodosaurus                | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_edmontonia                | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_camptosaurus              | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_pachycephalosaurus        | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_triceratops               | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_gallimimus_hatchling      | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_gallimimus_juvenile       | No  |                                      |                                          |
| DINOFOOD_C   | DinoMod       | mon_amargasaurus_juvenile     | No  |                                      |                                          |
</details>

I did not add the `CANPLAY` flag to monsters with `CATTLEFODDER` because in the code you can't play with them, hopefully someone in the future will add it to them so I can play with my pet cow Jhonny.

**These images aren't actually in-game, it's just an example**
acidic_zombear
zombie_dog

### CODE REVIEW

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 1]
```diff
@@ -348,6 +348,11 @@
     "reproduction": { "baby_egg": "egg_compsognathus", "baby_count": 3, "baby_timer": 12 },
     "baby_flags": [ "SPRING", "SUMMER" ],
     "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 3 },
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/dinosaur.json:353]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 2]
```diff
@@ -1147,7 +1150,12 @@
     "reproduction": { "baby_egg": "egg_dimorphodon", "baby_count": 3, "baby_timer": 24 },
     "baby_flags": [ "SPRING", "SUMMER" ],
     "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 3 },
-    "flags": [ "SEES", "SMELLS", "HEARS", "FLIES", "HIT_AND_RUN", "ANIMAL", "PATH_AVOID_DANGER_1", "CATFOOD" ],
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/dinosaur.json:1155]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 3]
```diff
@@ -1188,7 +1196,12 @@
     "description": "A huge feathered flying reptile, with a long pointed toothless beak and a long stiff neck.",
     "reproduction": { "baby_egg": "egg_quetzalcoatlus", "baby_count": 3, "baby_timer": 24 },
     "special_attacks": [ [ "scratch", 10 ] ],
-    "flags": [ "SEES", "SMELLS", "HEARS", "FLIES", "HIT_AND_RUN", "ANIMAL", "PATH_AVOID_DANGER_1", "CATFOOD", "PET_MOUNTABLE" ],
+    "flags": [ "SEES", "SMELLS", "HEARS", "FLIES", "HIT_AND_RUN", "ANIMAL", "PATH_AVOID_DANGER_1", "CANPLAY", "PET_MOUNTABLE" ],
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/dinosaur.json:1202]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/hatchling.json [hunk 4]
```diff
@@ -105,14 +105,19 @@
     "melee_cut": 1,
     "dodge": 1,
     "upgrades": { "age_grow": 14, "into": "mon_compsognathus" },
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/hatchling.json:110]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/hatchling.json [hunk 5]
```diff
@@ -386,13 +385,18 @@
     "description": "A tiny aquatic reptile hatchling with huge shiny eyes and pointy little teeth.",
     "default_faction": "mosasaurus_hatchling",
     "upgrades": { "age_grow": 30, "into": "mon_mosasaurus_juvenile" },
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/hatchling.json:390]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 6]
```diff
@@ -8,7 +8,12 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "magenta_green",
     "upgrades": { "age_grow": 65, "into": "mon_dilophosaurus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY", "SWIMS" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:13]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 7]
```diff
@@ -19,7 +24,12 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "magenta_green",
     "upgrades": { "age_grow": 90, "into": "mon_ceratosaurus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:29]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 8]
```diff
@@ -31,7 +41,12 @@
     "color": "red_white",
     "hp": 200,
     "upgrades": { "age_grow": 365, "into": "mon_spinosaurus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY", "SWIMS" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:46]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 9]
```diff
@@ -42,7 +57,12 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "brown_white",
     "upgrades": { "age_grow": 150, "into": "mon_allosaurus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:62]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 10]
```diff
@@ -53,7 +73,12 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "brown_white",
     "upgrades": { "age_grow": 365, "into": "mon_acrocanthosaurus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:78]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 11]
```diff
@@ -64,7 +89,12 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "brown_white",
     "upgrades": { "age_grow": 365, "into": "mon_siats" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:94]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 12]
```diff
@@ -76,7 +106,12 @@
     "color": "light_red_white",
     "dodge": 5,
     "upgrades": { "age_grow": 365, "into": "mon_albertosaurus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:111]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 13]
```diff
@@ -88,7 +123,12 @@
     "color": "light_red_white",
     "dodge": 6,
     "upgrades": { "age_grow": 365, "into": "mon_qianzhousaurus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:128]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 14]
```diff
@@ -101,7 +141,12 @@
     "color": "red_white",
     "dodge": 5,
     "upgrades": { "age_grow": 365, "into": "mon_tyrannosaurus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:146]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 15]
```diff
@@ -148,7 +194,12 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "red_green",
     "upgrades": { "age_grow": 14, "into": "mon_deinonychus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "KEENNOSE", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:199]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 16]
```diff
@@ -159,7 +210,12 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "dark_gray_white",
     "upgrades": { "age_grow": 80, "into": "mon_utahraptor" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATFOOD", "NO_BREED", "CANPLAY" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:215]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 17]
```diff
@@ -406,6 +452,11 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "light_gray_yellow",
     "upgrades": { "age_grow": 14, "into": "mon_pteranodon" },
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:457]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 18]
```diff
@@ -429,6 +480,11 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "light_gray_yellow",
     "upgrades": { "age_grow": 40, "into": "mon_quetzalcoatlus" },
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:485]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 19]
```diff
@@ -452,6 +508,11 @@
     "copy-from": "mon_gallimimus_juvenile",
     "color": "red_white",
     "upgrades": { "age_grow": 365, "into": "mon_mosasaurus" },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "CATFOOD", "NO_BREED", "CANPLAY", "SWIMS", "AQUATIC" ]
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with felines.",
```
Comment (actual-nh) [inline data/mods/DinoMod/monsters/juvenile.json:513]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with reptiles.",
```

**Review by actual-nh — COMMENTED** [review]
"CATFOOD" for dinosaurs, at least, does not indicate they're felines - it indicates they're obligate (and, for optimal health, almost pure) carnivores, like domestic cats. I will let @LyleSY comment on the other parts including petting responses.

**Comment by Saicchi** [comment]
My plan was to give them all the basic string first like all the other monsters then fix each mod individually to make things more organized. But doing it in this PR works too.

**Comment by PatrikLundell** [comment]
Dinosaurs aren't reptiles, but their own category (i.e. dinosaurs). If you have to cram them into any current day category it would have to be birds.

I would also expect different dinosaur species to have very different diets, as there were carnovors, herbivores, and, probably, omnivores. It makes about as much sense to give them all the same diet as it would to do the same with mammals.

**Comment by Saicchi** [comment]
I also had the idea of removing `feed` and `pet` strings from monsters from mod but I left them there so the mod contributors know where to look.

File: data/mods/DinoMod/monsters/dinosaur.json (hunk 1 above)
Comment (kevingranade) [inline data/mods/DinoMod/monsters/dinosaur.json:353]:
```suggestion
      "feed": "The %s seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell...",
```

**Comment by actual-nh** [comment]
> Dinosaurs aren't reptiles, but their own category (i.e. dinosaurs). If you have to cram them into any current day category it would have to be birds.

OK, "dinosaurs"? (Or if someone _else_ wishes to do ellipses - they're entirely too difficult to enter and edit for me.)

> I would also expect different dinosaur species to have very different diets, as there were carnovors, herbivores, and, probably, omnivores. It makes about as much sense to give them all the same diet as it would to do the same with mammals.

As I mentioned above:

>"CATFOOD" for dinosaurs, at least, does not indicate they're felines - it indicates they're obligate (and, for optimal health, almost pure) carnivores, like domestic cats.

"CATTLEFODDER" indicates a herbivore; "DOGFOOD" indicates an omnivore.

**Comment by Saicchi** [comment]
> "CATTLEFODDER" indicates a herbivore; "DOGFOOD" indicates an omnivore.
In this PR the name doesn't matter at all, it just keeps things organized. As long as they have at least one category in common, the food item can be used with the monster. I'll the food categories for dinosaurs and add them to their respective food items.

**Comment by actual-nh** [comment]
> I also had the idea of removing `feed` and `pet` strings from monsters from mod but I left them there so the mod contributors know where to look. Like changing `CATFOOD` to `DINOFOOD` or let the cyber mastiff puppy only eat cyber-cookies.

The latter is amusing; the former would not work properly because of variable diets, if done that way. Documentation of "CATFOOD" as simply indicating a carnivore could help, apparently.

**Comment by actual-nh** [comment]
For most carnivores anything that would give someone with a predator mutation pleasure to eat is _potentially_ edible, although too much fat will not be appetizing for most.

**Comment by actual-nh** [comment]
DOGFOOD, once the actual preferred diet is indicated, can be eliminated from most unless there's some reason to think they'd respond to a dog whistle - good point about that.

**Comment by actual-nh** [comment]
OK, that works re DinoMod; thanks! (I may do a PR rearranging the letters for which one is which among DINOFOOD_A, DINOFOOD_B, and DINOFOOD_C so it's DINOFOOD_C for carnivores, DINOFOOD_H for herbivores, and DINOFOOD_O for omnivores. Dunno whether a separate one for piscivores (fish-eaters) is needed.)

**Comment by Saicchi** [comment]
Both the monster and item can have any number of categories, and you can put the category in any comestible. So create as many categories as needed.

**Comment by PatrikLundell** [comment]
(Almost) exclusive fish eaters are likely to be aquatic or amphibian, and so not really candidates for taming.

**Review by LyleSY — CHANGES_REQUESTED** [review]
Eh, I'm going with Sea World rules for DinoMod. This implementation is much smarter than I understood, the flexibility is amazing. I made a suggestion to allow omnivores to eat all the things, and either Kevin or actual-nh's customizations seem fine to me. Really great stuff here.

**Comment by actual-nh** [comment]
> Sea World rules

?

**Comment by LyleSY** [comment]
SeaWorld trains killer whales and dolphins with treats. They aren't truly tame like a dog, but they'll perform commands. Granted those are both mammals. https://en.wikipedia.org/wiki/Captive_killer_whales#Training

**Review by LyleSY — APPROVED** [review]

**Review by Maleclypse — APPROVED** [review]
This is really impressive and the Aftershock section is complete.  Thank you!

**Pull request merged**